src/quo2/components/profile/profile_card/view.cljs

```diff
@@ -4,77 +4,102 @@
-           emoji-hash on-options-press show-emoji-hash? padding-bottom
-           show-options-button? show-user-hash? show-logged-in? on-card-press]
+  [{:keys [keycard-account? profile-picture name hash
+           customization-color emoji-hash on-options-press
+           show-emoji-hash? show-options-button? show-user-hash?
+           show-logged-in? on-card-press login-card? last-item? card-style]
     :or   {show-emoji-hash?     false
            show-user-hash?      false
            customization-color  :turquoise
            show-options-button? false
            show-logged-in?      false
-           key-card?            false}}]
-  [rn/touchable-without-feedback
-   {:on-press            on-card-press
-    :flex                1
-    :accessibility-label :profile-card}
-   [rn/view
-    (style/card-container
-     customization-color
-     (or padding-bottom (if show-emoji-hash? 12 10)))
-    [rn/view
-     {:style style/card-header}
-     [user-avatar/user-avatar
-      {:full-name         name
-       :profile-picture   profile-picture
-       :override-theme    :dark
-       :size              :medium
-       :status-indicator? false}]
-     [rn/view {:flex-direction :row}
-      (when show-logged-in?
-        [tag/tag
-         {:type                :icon
-          :size                32
-          :blurred?            true
-          :labelled?           true
-          :resource            :main-icons2/check
-          :accessibility-label :logged-in-tag
-          :icon-color          colors/success-50
-          :override-theme      :dark
-          :label               (i18n/label :t/logged-in)}])
-      (when show-options-button?
-        [button/button
-         {:size                32
-          :type                :blur-bg
-          :icon                true
-          :override-theme      :dark
-          :style               style/option-button
-          :on-press            on-options-press
-          :accessibility-label :profile-card-options}
-         :i/options])]]
-    [rn/view
-     {:style style/name-container}
-     [text/text
-      {:size            :heading-2
-       :weight          :semi-bold
-       :number-of-lines 1
-       :style           style/user-name} name]
-     (when key-card?
-       (icon/icon
-        :i/keycard
-        style/keycard-icon))]
-    (when show-user-hash?
-      [text/text
-       {:weight          :monospace
-        :number-of-lines 1
-        :style           style/user-hash} hash])
-    (when (and show-emoji-hash? emoji-hash)
-      [text/text
-       {:weight          :monospace
-        :number-of-lines 1
-        :style           style/emoji-hash} emoji-hash])]])
+           keycard-account?     false
+           login-card?          false
+           last-item?           false
+           card-style           {:padding-horizontal 20
+                                 :flex               1}}}]
+  [:f>
```

Comment:
Hey @Parveshdhull

I'm not sure this pattern is good in terms of performance and good practices :thinking: I guess the main component (which is just after the `:f>`) is recreated each time the component is mounted, since it's a function defined inside another function.

Also I think each time a property of the `profile-card` component is changed, then the re-render will recreate the function/component.

So what do you think about using the following pattern instead?:

```clojure
(defn profile-card* ;; <- not sure about this name, probably there're better ones
  [{:keys [
           ;; props
           ]
    :or   {
           ;; default props
           }}]
  (let [{:keys [width]}      (rn/use-window-dimensions) ;; <- the hook
        ;; other bindings
        ,,,
        ]
    ;; rest of the code
    ,,,
    ))

(defn profile-card [props]
  [:f> profile-card* props])

```

Reply:
changed, thank you very much!